src/internal_modules/roc_core/string_list.cpp

```diff
@@ -130,6 +130,28 @@ bool StringList::push_back(const char* str_begin, const char* str_end) {
     return true;
 }
 
+bool StringList::pop_back() {
+    if (size_ == 0) {
+        roc_panic("stringlist: list is empty");
+    }
+
+    const size_t blk_sz = back_->len;
+    if (!data_.resize(data_.size() - blk_sz)) {
+        return false;
+    }
+
+    size_--;
+    if (size_ > 0) {
+        back_ = (Header*)(data_.data() + data_.size() - blk_sz);
```

Comment:
Bug: here we subtract size of removed block, but instead we should get size of block before removed one and use it here.

Test passes because in test all strings have same lengths (3 chars). Please update test to use strings of different sizes.

Reply:
Thanks for pointing it out. I had completely missed it.
Fixed in recent commit with test cases for different sized strings.